curriculum/challenges/english/blocks/workshop-shortest-path-algorithm/687b45e16f38ad5f4e4bd799.md

```diff
@@ -0,0 +1,149 @@
+---
+id: 687b45e16f38ad5f4e4bd799
+title: Step 23
+challengeType: 20
+dashedName: step-23
+---
+
+# --description--
+
+Now that you've determined a node should be displayed, you need to format its path so it can be printed clearly.
+
+Inside the loop after the `continue` statement, create a variable called `path`. Assign it a string that joins all the node numbers in `paths[node_no]` using `" -> "` as the separator.
+
+Use a generator expression with `str()` to convert each node number to a string before joining them.
```

Comment:
I'm almost sure generator expressions are not mentioned during the entire curriculum. So we should either change this or teach them somewhere